### PR TITLE
Feat card component

### LINKED ISSUE #5237: [angular] Create a card component
We use `card` `card-body` classes everywhere. 
We should have a component as follows:

```html
<abp-card>
  <abp-title> Title </abp-title>
  <abp-body> Body </abp-body> <!-- abp-body could be omitted -->
<abp-card>
```

### PR DESCRIPTION
**Pull request by bariscanyilmaz**
### Description

Resolves #5237 

Created new card component to use instead of card class  

### Checklist

- [ ] I fully tested it as developer / designer and created unit / integration tests
- [ ] I documented it (or no need to document or I will create a separate documentation issue)

### CODE REVIEW

File: npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Component, ContentChild, Input } from '@angular/core';
+import { CardBodyComponent } from './card-body.component';
+import { CardTitleComponent } from './card-title.component';
+
+@Component({
+  selector: 'abp-card',
+  template: `
+  <div class="card" [ngClass]="class" [ngStyle]="style" >
+  <div class="card-title" *ngIf="cardTitleTemplate">
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts]: card-title component should constain 
` <div class="card-title" *ngIf="cardTitleTemplate">
    <ng-content select="abp-card-title"></ng-content>
  </div>`
  that  or it should be removed.

File: npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <ng-content select="abp-card-title"></ng-content>
+  </div>
+  <div class="card-body" *ngIf="cardBodyTemplate">
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts]: same for card-body

File: npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <ng-content select="abp-card-body"></ng-content>
+  </div>
+</div>`,
+})
+export class CardComponent {
+  @Input() class:string
+
+  @Input() style:string
+
+  @ContentChild(CardBodyComponent)
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts:17]: maybe use ng-content not component.

File: npm/ng-packs/packages/theme-shared/src/lib/theme-shared.module.ts [hunk 4]
```diff
@@ -52,6 +55,9 @@ const declarationsWithExports = [
   NgxDatatableListDirective,
   LoadingDirective,
   ModalCloseDirective,
+  CardComponent,
+  CardTitleComponent,
+  CardBodyComponent,
```
Comment (mahmut-gundogdu) [inline npm/ng-packs/packages/theme-shared/src/lib/theme-shared.module.ts]: I would do card.module.ts or card.standalone

**Review by mahmut-gundogdu — CHANGES_REQUESTED** [review]

File: npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  <div class="card-title" *ngIf="cardTitleTemplate">
+    <ng-content select="abp-card-title"></ng-content>
+  </div>
+  <div class="card-body" *ngIf="cardBodyTemplate">
+    <ng-content select="abp-card-body"></ng-content>
+  </div>
+</div>`,
+})
+export class CardComponent {
+  @Input() class:string
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts]: please change input name to 'cardClass'. Do not use native HTML attributes as input

File: npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  @Input() style:string
```
Comment (muhammedaltug) [inline npm/ng-packs/packages/theme-shared/src/lib/components/card/card.component.ts]: please change input name to 'cardStyle'. Do not use native HTML attributes as input

**Review by muhammedaltug — APPROVED** [review]

**Review by mahmut-gundogdu — APPROVED** [review]

**Pull request merged**